File: Timofeev PE-42/Bizzare Underground C++/Project1/Door.cpp
#include "Door.h"
#include <iostream>

Door::Door() : keyCounter(0) {
    std::cout << "--- Door Initialized ---" << std::endl;
    std::cout << "  Required keys: 3" << std::endl;
}

Door::~Door() {
    std::cout << "--- Door Destroyed ---" << std::endl;
}

void Door::doorOpen() {
    std::cout << "Door::doorOpen() called. Current key count: " << getKeyCounter() << std::endl;
    if (getKeyCounter() >= max_keys) {
        std::cout << "Door::doorOpen(): Opened successfully! Key count (" << getKeyCounter() << ") is sufficient." << std::endl;
    }
    else {
        std::cout << "Door::doorOpen(): Blocked. Need " << (max_keys - getKeyCounter()) << " more keys." << std::endl;
    }
}

int Door::getKeyCounter() const { return keyCounter; }
void Door::setKeyCounter(int count) {
    keyCounter = count;
    std::cout << "Door::setKeyCounter() called. New count: " << keyCounter << std::endl;
}

File: Timofeev-PE-42/Bizzare-Underground-Java/Door.java
public class Door { // Не наследуется от GameObject в этом примере
    private int keyCounter;

    public Door() {
        this.keyCounter = 0;
        System.out.println("--- Door Initialized ---");
        System.out.println("  Required keys: 3");
    }

    @Override
    protected void finalize() throws Throwable {
        System.out.println("--- Door Destroyed ---");
        super.finalize();
    }

    public void doorOpen() {
        System.out.println("Door.doorOpen() called. Current key count: " + getKeyCounter());
        if (getKeyCounter() >= 3) {
            System.out.println("Door.doorOpen(): Opened successfully! Key count (" + getKeyCounter() + ") is sufficient.");
        } else {
            System.out.println("Door.doorOpen(): Blocked. Need " + (3 - getKeyCounter()) + " more keys.");
        }
    }

    // Getter
    public int getKeyCounter() { return keyCounter; }

    // Setter
    public void setKeyCounter(int count) {
        this.keyCounter = count;
        System.out.println("Door.setKeyCounter() called. New count: " + keyCounter);
    }
}